### ISSUE TITLE
Transforming arc segment is not working properly

### ISSUE THREAD

**Issue by kpym** [fontello/svgpath#13]
The command 
```
svgpath("M0,0 A 10,10 45 0,1 10,10z").transform("scale(1,2)").toString() 
```
gives the wrong result 
```
"M0 0A10 20 45 0 1 10 20z"
```
It is obvious that if your ellipse has angle non 0 [mod pi/2] and you scale with two different values, you can't obtain the ellipse with the same angle.
You can check the result by comparing this resulting path with the transform made by the browser : 
```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="-10 -10 40 40">
  <!-- the original path -->
  <path transform="scale(1,2)" d="M0,0 A 10,10 45 0,1 10,10" style="fill:none;stroke-width:.1;stroke:blue"/>
  <!-- the path transformed by svgpath -->
  <path transform="" d="M 0 0 A 10 20 45 0 1 10 20" style="fill:none;stroke-width:.1;stroke:red"/>
</svg>
 ```
![arc_transform_test](https://cloud.githubusercontent.com/assets/3142701/11912604/3139e5f0-a645-11e5-8f34-73278c964051.png)

**Comment by puzrin** [comment]
Ups... tests fail too:

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="50 20 150 150">
  <!-- the original path -->
  <path transform="scale(2, 1.5)" d="M40 30a20 40 -45 0 1 20 50" style="fill:none;stroke-width:.1;stroke:blue"/>
  <!-- the path transformed by svgpath -->
  <path transform="" d="M80 45a40 60-45 0 1 40 75" style="fill:none;stroke-width:.1;stroke:red"/>
</svg>
```

https://github.com/fontello/svgpath/blob/master/test/api.js#L228-L238 . Don't know how it happened, because i remember that required to check visual equality.

/cc @rlidwka 

Arc transform code is here, if anyone wish to take a look https://github.com/fontello/svgpath/blob/master/lib/svgpath.js#L74-L127

**Comment by kpym** [comment]
I'll make a push request for an update ...

**Comment by puzrin** [comment]
Current algorithm is defective by design and need full rewrite (via circular arcs).

References:

1. http://stackoverflow.com/a/13102801/1031804 - lacks of invalid params normalization.
2. http://www.w3.org/TR/SVG/implnote.html#ArcCorrectionOutOfRangeRadii.
3. SVGO - works, but equations could be more easy to understand, with refer to w3.org terminology.

**Comment by puzrin** [comment]
Released 2.1.2 with your fix.

**Comment by kpym** [comment]
Thanks.
